### ISSUE TITLE
Set Printing All

### ISSUE THREAD

**Issue by thery** [coq-community/vscoq#81]
I don't know if it is the intended behaviour. But when executing `Set Printing All` the command is valid only on the current goal.

```
Goal 1 = 1 -> 2 = 1.
Set Printing All.
intro.
```

**Comment by Blaisorblade** [comment]
Right now the command gives a warning (IIRC) and suggests using a vscode command (display basic low-level...), so this might be closable.

**Comment by fakusb** [comment]
I find it very strange that setting printing flags does not effect Queries, e.g. "Print term.".

**Comment by Blaisorblade** [comment]
Flags do work, you must set them via IDE commands, like with CoqIDE — instead of `Set Printing All.`, use `> Display All Basic Low-level Contents.` (from the command palette)
A separate (and annoying) problem is that this works when running `Print term.` but not for `> Coq: Print` (or Cmd/Ctrl-Shift-P).

**Comment by fakusb** [comment]
Ok, that is a workaround for me, but it still is hard for an beginner if he encounters this in a demo file, where the author of the demo wants to show something about implicit arguments for example.

Is it a conscious design decision that "Set Printing" flags are not respected/"noticed" on the level of Vernacular-queries?

**Comment by thery** [comment]
Looking at implicit argument is such a frequent practice when proving in `Coq` that 
in a modern ide like `vscode` it would be nice to have a dedicated tool, such that 
a tool-tip when mousing-over term.

**Comment by maximedenes** [comment]
I agree there's a lot to improve there. Hopefully, it will be much easier to implement once we get #140.

**Comment by Blaisorblade** [comment]
@fakusb Good point on "demo files".

> Is it a conscious design decision that "Set Printing" flags are not respected/"noticed" on the level of Vernacular-queries?

That's consistent with CoqIDE, which _tells you_ to use the commands, so I assume it's intentional — but it might still be questionable.

**Comment by QinshiWang** [comment]
It seems `> Coq: Print Implicit Arguments` does not work properly. It should do `Set Printing Implicit` like CoqIDE, but it happens to open a menu.

**Comment by TheoWinterhalter** [comment]
It's `> Coq: Display Implicit Arguments` and it shows you a list of toggles, so I guess it does its job?

**Comment by QinshiWang** [comment]
@TheoWinterhalter Thanks for your reply. But implicit arguments refer to [link](https://coq.inria.fr/refman/language/extensions/implicit-arguments.html) (or [link](https://coq.inria.fr/refman/language/extensions/implicit-arguments.html#implicit-argument-binders) for examples). In CoqIDE, `Display Implicit Arguments` means to only display these implicit arguments, but not notations, coercions, etc. I hope VSCoq also has an option to display these arguments only.

**Comment by Blaisorblade** [comment]
@QinshiWang Selecting the "Implicit Arguments" toggle in the menu that appears should do the right thing. However, that a list of toggles appears is a bug — none of the other completions of `Coq: Display...` do that.

**Comment by TheoWinterhalter** [comment]
Ah ok. I thought this was actually nice.

**Comment by fakusb** [comment]
For me, pressing enter ob the popping-up menu does also not do anything. But it works for yoy, theo?

**Comment by TheoWinterhalter** [comment]
Yes, it seems to work.

**Comment by QinshiWang** [comment]
@fakusb It seems also to do nothing for me. I am using VSCoq 0.3.2 from VS Code extension market.

**Comment by TheoWinterhalter** [comment]
Ok, it doesn't work for all of the toggles indeed. "Notation" works for me, but not "Implicit Arguments".

**Comment by fakusb** [comment]
I'm firing up my debugger and tryu to fix this

**Comment by fakusb** [comment]
That other issue is handled in #182. I have a build of my forks current master-branch, including other improvements, at https://github.com/fakusb/vscoq/releases/tag/0.3.3-alpha4-custom

**Comment by maxild** [comment]
I am a self-learner/beginner going through `Basics.v` from SF (first book)...

When I use `Check` on a defintion I would like to see the result in pretty-printed notation. Instead I see the actual low level terms. That is would like to see

```coq
Check mult_n_O.
    : forall n : nat, 0 = n * 0   
``

Instead I see

```coq
Check mult_n_O.
    : forall n : nat, @eq nat O (Nat.mul n O)
```

I have tried `Set Printing All.` in the coq/gallina source, and various "> Coq: Display..." commands in the vscode IDE, but haven't found a solution. Using coqtop in my terminal works as expected.

Is this specific to VsCoq (and due to this issue)?

**Comment by SkySkimmer** [comment]
Set Printing All is the opposite of what you want.

**Comment by maxild** [comment]
> Set Printing All is the opposite of what you want.

Really?

```
Coq < Unset Printing Notations.

Coq < Check ((0 + 1) + 1) : nat.
Nat.add (Nat.add 0 1) 1 : nat
     : nat

Coq < Set Printing Notations.

Coq < Check ((0 + 1) + 1) : nat.
0 + 1 + 1 : nat
     : nat
```

```
Coq < Unset Printing All.

Coq < Check ((0 + 1) + 1) : nat.
0 + 1 + 1 : nat
     : nat

Coq < Set Printing All.

Coq < Check ((0 + 1) + 1) : nat.
Nat.add (Nat.add O (S O)) (S O) : nat
     : nat
```

So `Printing All` works the opposite compared to `Printing Notation`...weird

**Comment by TheoWinterhalter** [comment]
For me vscoq always prints notation unless I tell it not to.
`Set Printing All` is not to be understood as using all options but as printing everything that the term might be hiding so this removes notations, shows coercions and implicit arguments, etc.

**Comment by maxild** [comment]
> Set Printing All is not to be understood as using all options but as printing everything that the term might be hiding so this removes notations, shows coercions and implicit arguments, etc.

Okay thanks. Consistent with above. I probably (blindly) executed vscoq/IDE commands for `>Coq: Display All Low Level Contents` (guess this is eq to `Set Printing All`) and `Coq: Display Notations` (guess this is eq to `Set Printing Notation`), and now I know, that those 2 work opposite wrt notations. After restaring vscode notations are printed.

Is the preferred way to "toggle" printing settings using the vscoq IDE commands, and where can I see the current settings? (just tried using coqide, and it has a view menu with visible radio/toggle widgets)

**Comment by Blaisorblade** [comment]
> Is the preferred way to "toggle" printing settings using the vscoq IDE commands

Yes

> and where can I see the current settings?

Good question 👍

**Comment by andreiburdusa** [comment]
I'm working through `Imp.v` from Software Foundations and `Coq> Display Implicit Arguments` seems to have no effect.
Interpreting the line
```
Check @ceval_example2.
``` 
always gives
```
ceval_example2
	 : empty_st =[ X := 0; Y := 1; Z := 2 ]=> (Z !-> 2; Y !-> 1; X !-> 0)
```
Also, interpreting the line `Set Printing Implicit.` has no effect either, but it looks like that's expected from what I read above.

**Comment by Blaisorblade** [comment]
@andreiburdusa Your example is printed as a notation, so the `Implicit` setting will not have an effect; you must first disable notations via `Coq: Display Notations`, and _then_ the `Implicit` setting will have an effect.

**Comment by Blaisorblade** [comment]
Upstream issue: https://github.com/coq/coq/issues/16080.

And https://github.com/coq/coq/issues/5178 is a different but related one.
